vncviewer/vncviewer.cxx

```diff
@@ -109,28 +108,47 @@ static const char *about_text()
   return buffer;
 }
 
-void exit_vncviewer(const char *error, ...)
+
+void abort_vncviewer(const char *error, ...)
 {
   // Prioritise the first error we get as that is probably the most
   // relevant one.
-  if ((error != NULL) && (exitError == NULL)) {
+  char* errorString = NULL;
+  if (error != NULL) {
     va_list ap;
 
     va_start(ap, error);
-    exitError = (char*)malloc(1024);
-    if (exitError)
-      (void) vsnprintf((char*)exitError, 1024, error, ap);
+    errorString = (char*)malloc(1024);
+    if (errorString) {
+      (void) vsnprintf((char*)errorString, 1024, error, ap);
+      vlog.error("%s",errorString);
+    }
     va_end(ap);
   }
 
-  if (inMainloop)
-    exitMainloop = true;
-  else {
-    // We're early in the startup. Assume we can just exit().
-    if (alertOnFatalError && (exitError != NULL))
-      fl_alert("%s", exitError);
-    exit(EXIT_FAILURE);
+  exit(1);
+}
+
+
+void exit_vncviewer(const char *error, ...)
```

Comment:
This should probably be renamed given that it no longer reliably exits vncviewer.

Reply:
Referring to "vncviewer" might no longer be relevant for this function as it is now more about terminating the current connection, not the entire viewer.